src/main/java/no/nav/bilag/BrevserverConsumer.java

```diff
@@ -0,0 +1,87 @@
+package no.nav.bilag;
+
+import lombok.extern.slf4j.Slf4j;
+import no.nav.bilag.auth.OboTokenService;
+import no.nav.bilag.exceptions.BrevserverFunctionalException;
+import no.nav.bilag.exceptions.BrevserverTechnicalException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+
+import static java.lang.String.format;
+import static org.springframework.http.HttpHeaders.CONTENT_DISPOSITION;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_PDF_VALUE;
+
+@Slf4j
+@Component
+public class BrevserverConsumer {
+
+	private final WebClient webClient;
+	private final OboTokenService oboTokenService;
+
+	public BrevserverConsumer(BilagProperties bilagProperties,
+							  OboTokenService oboTokenService,
+							  WebClient webClient) {
+		this.oboTokenService = oboTokenService;
+		this.webClient = webClient.mutate()
+				.baseUrl(bilagProperties.getEndpoints().getBrevserver().getUrl())
+				.build();
+	}
+
+	private void setContentDisposition(HttpHeaders headers, Long dokId) {
+		headers.add(CONTENT_TYPE, APPLICATION_PDF_VALUE);
+		headers.add(CONTENT_DISPOSITION, format("inline; filename=\"OEBS_%s.pdf\"", dokId));
+	}
+
+	private void setAuthorization(HttpHeaders headers, String accessToken) {
+		headers.setBearerAuth(oboTokenService.fetchAccessToken(accessToken));
+	}
+
+	public byte[] hentDokument(Long dokId, String accessToken) {
+		log.info("hentDokument henter dokument med dokId={} fra brevserver", dokId);
+
+		var response = webClient.get()
+				.uri(uriBuilder -> uriBuilder
+						.path("/{dokId}")
+						.build(dokId))
+				.headers(headers -> {
+					setContentDisposition(headers, dokId);
```

Comment:
Husk oppsett for å hente større filer enn default i webclient
Se https://github.com/navikt/safselvbetjening/pull/124/files

Reply:
Goodie! Då er det med